### DISCUSSION TITLE
Why not use this package (package:googleapis_auth) with Flutter applications?

### DISCUSSION THREAD

**Discussion by kjxbyz** [google/googleapis.dart#616] [Q&A] [answered]
So how do I use the `API Key` generated in google console to call the YouTube API in the flutter application?

**Comment by kevmoo** [comment]
Check out https://www.youtube.com/watch?v=z4MsuZiEezY

**Comment by mattrltrent** [comment, marked as answer]
@kjxbyz 

# (1) If you need to use `googleapis_auth`

If you want to use [googleapis_auth](https://pub.dev/packages/googleapis_auth), don't. They say use [extension_google_sign_in_as_googleapis_auth](https://pub.dev/packages/extension_google_sign_in_as_googleapis_auth) instead directly in the package [documentation](https://pub.dev/packages/googleapis_auth#:~:text=Use%20package%3Aextension_google_sign_in_as_googleapis_auth%20instead).

# (2) If you just want to use just your API key and raw HTTP

If you don't want to use a package, since you said you have your own API key, you can simply use the [http](https://pub.dev/packages/http) package instead by plugging in your `API KEY` variable to requests to Google's endpoint directly. 

# (3) The easiest way using an abstraction package with your API key

For *even easier usage*, consider the [youtube_api](https://pub.dev/packages/youtube_api) Flutter package. This basically abstracts away any complexities. Here's an example [blog post](https://swaroop-sambhayya.medium.com/how-to-access-and-get-the-youtube-data-through-api-in-flutter-ec61160b673f) of this [youtube_api](https://pub.dev/packages/youtube_api) package being used.

Here is a simple code snippet from it showing just how simple it is to use (credit to the blog post):

```dart
YoutubeAPI yt = YoutubeAPI(api_key, maxResults: 6, type: "video");
yt.search("video you want to search for") //search video
yt.getTrends(regionCode:"Your country's apha2 region code" ) //get trending videos
yt.nextPage() //get the results of next page
yt.previousPage() //get the results of previous page
yt.video(["videoId1","videoId2"]) //get the video data based on list given in parameter
yt.channel(["channelid1","channelid2"]) //get channel data based on list provided 
```
